### DISCUSSION TITLE
licznik firmowy

### DISCUSSION THREAD

**Discussion by olekmarcin2** [PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus#154] [Q&A] [answered]
Czy jest jakieś ograniczenie dla liczników Firmowych lub taryfy C12 ?
mam na koncie Tauron Elicznik trzy PPE dwa prywatne i jeden firmowy prywatne się dodają bez problemu
natomiast firmowy jest na liście ale podczas dodawania jest komunikat login to failed
PPE prywatne zaczyna sie literami DO  
PPE firmowy  WO
Jet na to jakiś sposób ?

**Comment by PiotrMachowski** [comment]
Nie miałem do czynienia z licznikami firmowymi, więc nie jestem w stanie zbyt dużo powiedzieć. Prawdopodobnie coś pod spodem się może delikatnie różnić, ale ciężko by mi było przeanalizować sytuację bez dostępu do konta

**Reply by olekmarcin2** [reply]
ok mogę podać dane do logowania sprawdziłem dodatkowo na innym koncie i jest tak samo.
proszę dane priv gdzie podać dane do logowania

**Reply by PiotrMachowski** [reply]
Możesz wysłać na `piotr.machowski.dev [at] gmail.com`

**Comment by karpiq24** [comment]
Nie działa konkretnie pobieranie konfiguracji z metody calculate_configuration w connector.py. Trochę pogrzebałem i przy takich zmianach pobieranie konfiguracji już działa:

```
index 2501b96..7a0c7da 100644
--- a/custom_components/tauron_amiplus/connector.py
+++ b/custom_components/tauron_amiplus/connector.py
@@ -319,7 +319,7 @@ class TauronAmiplusConnector:
         }
         self.log(f"Downloading daily data for day: {day_str}, generation: {generation}")
         values = self.get_chart_values(payload)
-        if values is not None and not any(a is None for a in values['data']['values']):
+        if values:
             self.add_all_data(values, day)
             self._cache.add_value(day, generation, values)
             self.log(f"Downloaded daily data for day: {day_str}, generation: {generation}")
@@ -346,6 +346,7 @@ class TauronAmiplusConnector:
         return post
 
     def get_chart_values(self, payload):
+        payload["energy"] = "1"
         return self.execute_post(CONST_URL_ENERGY, payload)
 
     def execute_post(self, url, payload):
```

Przy firmowym liczniku wszystkie wartości w `values['data']` to `None`, jak wejdę na podstronę z licznikiem to przeglądarka robi zapytania do api: `/energia/wo/api` natomiast w kodzie używane jest `/energia/api`. Przy podmianie w kodzie tego adresu, wartości są pobierane prawidłowo ale to api nie zwraca taryfy, która jest potrzebna w `calculate_configuration`.

Nie wiem czy coś jeszcze w dalszych krokach się psuje przy firmowym liczniku, nie za bardzo mam pojęcie jak debugować home assistanta, dopiero zaczynam w nim grzebać. Ale może to co znalazłem pomoże wdrożyć poprawkę?

**Reply by PiotrMachowski** [reply]
Wydaje mi się, że jakiś czas temu integracja działała również dla firmowych liczników. Bym musiał na to jeszcze raz rzucić okiem.

**Comment by Raku2008** [comment]
Witam czy problem z licznikami firmowymi został rozwiązany? Posiadam licznik w taryfie  C12 i przy próbie logowania dostaje komunikat login to failed

**Reply by PiotrMachowski** [reply]
Czy na stronie eLicznika po zalogowaniu jest widoczny adres email? Jeśli tak, to czy jest on podany małymi, czy wielkimi literami?

**Reply by Raku2008** [reply]
Tak jest widoczny jest podany małymi literami dokładne tak samo jak drugi licznik który mam w taryfie G11 i z tym nie ma problemu z dodaniem do HA
Dodam że po wprowadzeniu danych do logowania pojawia się okno wyboru punktu poboru i tu jest wszystko ok jest widoczny licznik do którego chce się zalogować ale po zatwierdzeniu pojawia się "Failed to login"

**Comment by olekmarcin2** [comment]
Witam przez weekend potestuję mam  konta firmowe C12A drugie C21 i jedno mieszane G12,G11,C21
to ostatnie nie pozwalało dodać licznika C21. Kilka dni temu była aktualizacja e-licznik

**Reply by Raku2008** [reply]
Coś udało się ustalić do do taryfy C12A?

**Comment by olekmarcin2** [comment]
Witam przepraszam za Timing ale sprawy zawodowe zabieraja max czasu.. 
Potestowałem kilka kont firmowych: 
taryfa C12A dwukierunkowy działa
taryfa G11W dwukierunkowy działa (nie wiem jak odczytać poszczególne taryfy- wszystko mam razem)
taryfa C21 jednokierunkowy amiplus widzi PPE i nazwę ale daje komunikat Failed to login
taryfa C21 dwukierunkowy  amiplus widzi PPE i nazwę ale daje komunikat Failed to login

wyśle na prv dane logowania do C21

**Reply by PiotrMachowski** [reply]
> taryfa G11W dwukierunkowy działa (nie wiem jak odczytać poszczególne taryfy- wszystko mam razem)

Wartości dla taryf są dostępne w atrybutach

> wyśle na prv dane logowania do C21

Dzięki! To na pewno bardzo pomoże

**Comment by PiotrMachowski** [comment]
Dzięki danym udostępnionych przez @olekmarcin2 udało mi się pobrać dane dla liczników firmowych, także prawdopodobnie niedługo ich obsługa będzie dodana do integracji

**Comment by olekmarcin2** [comment]
Taryfa c21 dotyczy umowy powyżej 40Kw czyli dla dużych odbiorców.
Taryfa c12 działała poprawnie

**Comment by Raku2008** [comment]
U mnie przy taryfie c12a podczas dodawania do HA  "Failed to login"

**Comment by PiotrMachowski** [comment]
Czy w wersji [v2.6.0-beta](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus/releases/tag/v2.6.0-beta) wszystko działa?

**Comment by olekmarcin2** [comment]
Sprawdzę jutro

Dnia 13 stycznia 2024 20:49:58 CET, Piotr Machowski ***@***.***> napisał/a:
>Czy w wersji [v2.6.0-beta](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus/releases/tag/v2.6.0-beta) wszystko działa?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus/discussions/154#discussioncomment-8120014
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Wysłane za pomocą K-9 Mail.

**Comment by PiotrMachowski** [comment, marked as answer]
Liczniki firmowe są wspierane od wersji [v2.6.0](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus/releases/tag/v2.6.0)
